### PR TITLE
Update minimum password length requirements

### PR DESCRIPTION
**Pull request by techknowlogick**

### CODE REVIEW

**Comment by wxiaoguang** [comment]
I do not think it really makes sense, for example: 12345678 perfectly matches `MIN_PASSWORD_LENGTH = 8`

To make the password secure, it needs more rules.

**Comment by KN4CK3R** [comment]
Rules aren't really helping. We could add a "password strength meter" like https://github.com/dropbox/zxcvbn to show how secure a password is.

> Reasonable Password Policy Example
> - Passwords must be at between 12 and 4,096 characters in length.
> - Passwords can contain any characters (including Unicode).
> - We strongly encourage the use of a password manager to generate and store your passwords.
> - Your password strength must be at least level 3 (on the 0-4 scale).
>
> That's it. Don't tell people what their password can or cannot contain. Don't refuse longer passwords. Do stop people from shooting themselves in the foot, but don't interfere beyond what's necessary to prevent foot-bullets.

**Comment by wxiaoguang** [comment]
IMO the "password strength meter" is also "rule-based" 🤣 

And one more thing, even if some passwords are likely safe, but they are already in the password dict, so sometimes pwned-check is also necessary (well, it depends ....)

But I think making the password checking too complex is not in Gitea's scope, in most cases, SSO and 2FA are better solutions.

**Comment by KN4CK3R** [comment]
Yes, just want to prevent rules like "must contain 2 lower, 3 middle and 4 upper case letters". Then as user you think "wtf are middle letters?".

File: increase_min_pass_length.diff [hunk 1]
```diff
@@ -0,0 +1,39 @@
+diff --git a/custom/conf/app.example.ini b/custom/conf/app.example.ini
```
Comment (lafriks) [inline increase_min_pass_length.diff]: why is this file here? 🤔

Comment (techknowlogick) [reply increase_min_pass_length.diff]: Thanks for catching this. I copied the patch from another machine to create the commit and push it, and forgot to rm this file.

**Review by lafriks — APPROVED** [review]

**Review by silverwind — APPROVED** [review]
Can do more later

**Comment by silverwind** [comment]
e2e test failure may be related.

**Comment by techknowlogick** [comment]
@silverwind thanks :) I've updated e2e tests, and they are now passing.
